### ISSUE TITLE
Sub-directories in content

### ISSUE THREAD

**Issue by robfig** [stackb/rules_hugo#10]
Using a standard hugo_site: 

```
hugo_site(
    name = "hugo",
    config = "config.yaml",
    content = glob(["content/**"]),
    static = glob(["static/**"]),
    layouts = glob(["layouts/**"]),
    theme = "@com_github_alex_shpak_hugo_book//:book",
)
```

I'm finding that this doesn't work with multiple levels under content:

```
content/ 
  docs/
    index.md
    basics/
      overview.md
```

The reason is that Bazel removes the directory structure, and by the time Hugo runs on it, the files look like this:

```
content/ 
  index.md
  overview.md
```

Any ideas for how to fix this?

Thank you!
Rob

**Comment by robfig** [comment]
Oh, I was confusing this issue with a completely unrelated one I had run into before with pkg_tar. This should be a straightforward fix to hugo_site.bzl:copy_to_dir. I'll submit a PR.

**Comment by robfig** [comment]
It seems like this is sort of a big issue; it uses `ctx.actions.declare_directory()` to capture all of the files that hugo generates, but it appears that it only gets the files in that immediate directory, but no sub-directories. After I corrected it for content, I see that my images are being omitted for the same reason.

**Comment by robfig** [comment]
I was wrong about that. It was actually due to running hugo in a faraway directory, and the flags passed were not sufficient to correct that.

I've solved my problem here:
https://github.com/robfig/rules_hugo/commit/40aeae326ac3e362ff27748ba39894c4f61fc84c

.. but it seems like it doesn't fit well with the current rules. I think that if we're using --source under the hood, it might make sense to have the rules accept all the source files in a single "srcs" parameter rather than split out.

**Comment by pcj** [comment]
Sounds like a good change.  Do you have a full example with the file layout?

**Comment by pcj** [comment]
Hi @robfig I like that theme and your changes. I have your changes plus a new repository rule on the [modernize](https://github.com/stackb/rules_hugo/tree/modernize) branch.  Is it cool with you to merge this to master, with an attribution for your contribution?

**Comment by robfig** [comment]
👍 
Be my guest!

**Comment by pcj** [comment]
Closed by #11
